plotly_resampler/figure_resampler/figure_resampler_interface.py

```diff
@@ -719,15 +652,16 @@ def _parse_get_trace_props(
             # when y argument is used for the trace constructor instead of hf_y), we
             # transform it to type string as such it will be sent as categorical data
             # to the downsampling algorithm
-            if hf_y.dtype == "object":
+            if hf_y.dtype == "object" or hf_y.dtype.type == np.str_:
                 # But first, we try to parse to a numeric dtype (as this is the
                 # behavior that plotly supports)
                 # Note that a bool array of type object will remain a bool array (and
                 # not will be transformed to an array of ints (0, 1))
                 try:
                     hf_y = pd.to_numeric(hf_y, errors="raise")
                 except ValueError:
-                    hf_y = hf_y.astype("str")
+                    # TODO: would be great if we could omit pandas here
+                    hf_y = pd.Series(data=hf_y, copy=False, dtype="category").values
```

Comment:
Alternative would be using the `pd.Categorical` constructor (which is the output of the `.values`)

![image](https://user-images.githubusercontent.com/18898740/232236538-f5057961-fa43-4164-9fd7-0f5d3b85bbba.png)

Not really sure whether this is copy / or non-copy :upside_down_face: 
But shows +/- the same performance :handshake: 

![image](https://user-images.githubusercontent.com/18898740/232238478-e55d65ac-c2c6-41f5-9bf3-84ee2f9f4bb1.png)

Reply:
![image](https://user-images.githubusercontent.com/38005924/233849936-4cd5b8ba-c0f4-487d-866e-6196e962d37f.png)
as can be seen from the above tests ; the `pd.Categorical` and `pd.Series` construction both have the same memory and time-overhead.

Reply:
I would switch to `pd.Categorical` constructor then

Reply:
okay!